neuro_san/internals/graph/persistence/mcp_clients_info_restorer.py

```diff
@@ -0,0 +1,92 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+import os
+from typing import Any, Dict
+
+import json
+from pyparsing.exceptions import ParseException
+from pyparsing.exceptions import ParseSyntaxException
+
+from leaf_common.persistence.easy.easy_hocon_persistence import EasyHoconPersistence
+from leaf_common.persistence.interface.restorer import Restorer
+
+from neuro_san import DEPLOY_DIR
```

Comment:
FWIW: DEPLOY_DIR within the neuro_san repo is neuro_san/deploy
This is where all the Dockerfile-related stuff lives. Are you sure this is where you want to look for the mcp_info.hocon file?